src/auto-evo/simulation/MichePopulation.cs

```diff
@@ -213,15 +213,12 @@ private static void SimulatePatchStep(SimulationConfiguration simulationConfigur
             return;
 
         var leafNodes = new List<Miche>();
-        miche.GetLeafNodes(leafNodes, x => x.Occupant != null);
+        miche.GetLeafNodes(leafNodes, x => x.Occupant is MicrobeSpecies);
```

Comment:
I'm pretty worried about this change because once we add other species types to auto-evo I reckon this might be forgotten and wreak havoc. What's the reason behind this change?

Reply:
Reading the later code I now think I understand this but this should get a pretty big TODO and warning slapped on that this will be majorly wrong once other than microbe species can be fed into auto-evo.

Reply:
We need to make sure that a later call to the return from this is not null or not a microbe species. I && the microbe species check to this, but the lint wanted me to simplify it. Autoevo is currently littered with checks like this as Multicell species are partially supported. I doubt that this specific check would be missed as there are so many other like it I would probably just crtl + f "is MicrobeSpecies".

Reply:
Could you still add the TODO comment for my piece of mind here?
In most cases in my head it is that Species is usually used but just before microbe specific stuff is done it is then checked, but in this situation the other species types are pruned from the data much earlier.

Reply:
I can add that TODO